agent/proxycfg/snapshot.go

```diff
@@ -735,9 +735,12 @@ type configSnapshotAPIGateway struct {
 	// UpstreamsSet is the unique set of UpstreamID the gateway routes to.
 	UpstreamsSet routeUpstreamSet
 
-	HTTPRoutes   watch.Map[structs.ResourceReference, *structs.HTTPRouteConfigEntry]
-	TCPRoutes    watch.Map[structs.ResourceReference, *structs.TCPRouteConfigEntry]
-	Certificates watch.Map[structs.ResourceReference, *structs.InlineCertificateConfigEntry]
+	HTTPRoutes watch.Map[structs.ResourceReference, *structs.HTTPRouteConfigEntry]
+	TCPRoutes  watch.Map[structs.ResourceReference, *structs.TCPRouteConfigEntry]
+
+	// TODO(nathancoleman) Consider consolidating the 2 below into a map of structs.ConfigEntry
```

Comment:
@nathancoleman Did you want this TODO to live on after this is merged?

Reply:
I was considering combining these into a single watch map; however, that kinda breaks the patterns that we use everywhere with watch map and config entry type being 1:1. I'll remove the TODO